From 6a85c7d6c5030a09f84ffa7cd536761debfc6793 Mon Sep 17 00:00:00 2001
From: Mike Looijmans <mike.looijmans@topic.nl>
Date: Wed, 2 Jul 2014 13:08:26 +0200
Subject: [PATCH] Add support for topic-miami

topic-miami board is based on the zynq-miami prototype boards.
---
 board/xilinx/dts/topic-miami.dts |   14 ++++++++++
 boards.cfg                       |    1 +
 include/configs/topic_miami.h    |   55 ++++++++++++++++++++++++++++++++++++++
 3 files changed, 70 insertions(+)
 create mode 100644 board/xilinx/dts/topic-miami.dts
 create mode 100644 include/configs/topic_miami.h

diff --git a/board/xilinx/dts/topic-miami.dts b/board/xilinx/dts/topic-miami.dts
new file mode 100644
index 0000000..1744b4f
--- /dev/null
+++ b/board/xilinx/dts/topic-miami.dts
@@ -0,0 +1,14 @@
+/*
+ * Topic Miami board DTS
+ *
+ * Copyright (C) 2014 Topic Embedded Products
+ *
+ * SPDX-License-Identifier:	GPL-2.0+
+ */
+/dts-v1/;
+#include "zynq-7000.dtsi"
+
+/ {
+	model = "Zynq Miami Board";
+	compatible = "xlnx,topic-miami", "xlnx,zynq-7000";
+};
diff --git a/boards.cfg b/boards.cfg
index 76a40c5..2ef3e9a 100644
--- a/boards.cfg
+++ b/boards.cfg
@@ -356,6 +356,7 @@ Active  arm         armv7          socfpga     altera          socfpga
 Active  arm         armv7          u8500       st-ericsson     snowball            snowball                             -                                                                                                                                 Mathieu Poirier <mathieu.poirier@linaro.org>
 Active  arm         armv7          u8500       st-ericsson     u8500               u8500_href                           -                                                                                                                                 -
 Active  arm         armv7          vf610       freescale       vf610twr            vf610twr                             vf610twr:IMX_CONFIG=board/freescale/vf610twr/imximage.cfg                                                                         Alison Wang <b18965@freescale.com>
+Active  arm	    armv7	   zynq	       xilinx	       zynq	   	   topic_miami				-                                                                                                                                 Mike Looijmans <mike.looijmans@topic.nl>
 Active  arm	    armv7	   zynq	       xilinx	       zynq	   	   zynq_zc70x				-                                                                                                                                 Michal Simek <monstr@monstr.eu>:Jagannadha Sutradharudu Teki <jaganna@xilinx.com>
 Active  arm	    armv7	   zynq	       xilinx	       zynq	   	   zynq_zed				-                                                                                                                                 Michal Simek <monstr@monstr.eu>:Jagannadha Sutradharudu Teki <jaganna@xilinx.com>
 Active  arm	    armv7	   zynq	       xilinx	       zynq	   	   zynq_microzed			-                                                                                                                                 Michal Simek <monstr@monstr.eu>:Jagannadha Sutradharudu Teki <jaganna@xilinx.com>
diff --git a/include/configs/topic_miami.h b/include/configs/topic_miami.h
new file mode 100644
index 0000000..a1a4f21
--- /dev/null
+++ b/include/configs/topic_miami.h
@@ -0,0 +1,55 @@
+/*
+ * (C) Copyright 2014 Topic Embedded Products
+ *
+ * Configuration for Zynq Evaluation and Development Board - Miami
+ * See zynq-common.h for Zynq common configs
+ *
+ * SPDX-License-Identifier:	GPL-2.0+
+ */
+
+#ifndef __CONFIG_TOPIC_MIAMI_H
+#define __CONFIG_TOPIC_MIAMI_H
+
+#define CONFIG_SYS_SDRAM_SIZE		(512 * 1024 * 1024)
+#define CONFIG_ZYNQ_PS_CLK_FREQ		33333333UL
+#define CONFIG_ZYNQ_SERIAL_UART0
+
+
+/* No NOR flash */
+#define CONFIG_SYS_NO_FLASH
+/* SD and QSPI support. SD/USB is only availabe with a carrier board. */
+#define CONFIG_ZYNQ_SDHCI0
+#define CONFIG_ZYNQ_QSPI
+/* #define CONFIG_ZYNQ_USB */
+
+/* NAND SPL support for zynq is not available yet! */
+#ifdef ZYNQ_NAND_SPL_IMPLEMENTED
+# define CONFIG_NAND_ZYNQ
+# define CONFIG_SYS_NAND_PAGE_SIZE	2048
+# define CONFIG_SYS_NAND_U_BOOT_OFFS 0x20000
+# define CONFIG_SPL_NAND_INIT
+# define CONFIG_SPL_NAND_BOOT
+# define CONFIG_SPL_NAND_SUPPORT
+#endif
+
+#define CONFIG_DEFAULT_DEVICE_TREE	topic-miami
+
+/* Don't retrieve environment (quicker boot) */
+#define CONFIG_ENV_IS_NOWHERE
+
+/* I2C test stuff */
+#define CONFIG_SYS_I2C_ZYNQ
+
+#include "zynq-common.h"
+
+/* No FPGA support required */
+#undef CONFIG_FPGA
+#undef CONFIG_FPGA_XILINX
+#undef CONFIG_FPGA_ZYNQPL
+#undef CONFIG_CMD_FPGA
+#undef CONFIG_FPGA_LOADFS
+
+/* Extras */
+#define CONFIG_CMD_MEMTEST
+
+#endif /* __CONFIG_TOPIC_MIAMI_H */
-- 
1.7.9.5

